tests/chainermn_tests/communicator_tests/test_communicator.py

```diff
@@ -281,6 +339,105 @@ def check_send_recv(param, use_gpu):
     check_send_and_recv_tuple(communicator, data)
 
 
+class SimpleMock(object):
+    """A very simple mock class.
+
+    To test mixed-precision communication, it is necessary to check
+    the data type of actual communication, as well as check the results.
+    This is a very simple mock class to intercept the allreduce
+    function call to the underlying actual communicators, such as
+    PureNcclCommunicator and FlatCommunicator.
+
+    NOTE: unittest.mock is from PYthon 3.3.
+    """
+
+    def __init__(self, orig_obj):
+        self.orig_obj = orig_obj
+        self.__mock_call_args = []
+        self.__mocks = {}
+
+    def __getattr__(self, attr):
+        if attr == '__mock_call_args':
+            return self.__mock_call_args
+        if attr not in self.__mocks:
+            self.__mocks[attr] = SimpleMock(getattr(self.orig_obj, attr))
+        return self.__mocks[attr]
+
+    def __call__(self, *args, **kwargs):
+        self.__mock_call_args.append([args, kwargs])
+        self.orig_obj.__call__(*args, **kwargs)
+
+
+def check_allreduce_grad_mixed_dtype(param, model, use_gpu):
+    # Checks the actual allreduce communication is performed
+    # in the correct data type (FP16 or FP32)
+
+    chainer.global_config.dtype = param.global_dtype
```

Comment:
`chainer.global_config` is a global variable. 
So, you need to restore default value to it.

I think that it is better to use `with` statement like this:
https://github.com/shu65/chainer/blob/12cf11b39b69a6a40f168f4611586cd49e9a8e07/tests/chainermn_tests/links_tests/test_batch_normalization.py#L105

Reply:
I misunderstood the usage of `chainer.get_dtype`. 
I changed `chainer.get_dtype(chainer.global_config.dtype)` --> `chainer.get_dtype()` and seems working. Thanks!